components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/IdentityKeyStoreResolver.java

```diff
@@ -530,10 +530,25 @@ private String getCustomKeyStoreConfig(String keyStoreName, String configName)
 
     private void parseIdentityKeyStoreMappingConfigs() {
 
-        OMElement keyStoreMappingsElem = IdentityConfigParser.getInstance().getConfigElement(
-                IdentityKeyStoreResolverConstants.CONFIG_ELEM_SECURITY).getFirstChildWithName(
-                IdentityKeyStoreResolverUtil.getQNameWithIdentityNameSpace(
-                        IdentityKeyStoreResolverConstants.CONFIG_ELEM_KEYSTORE_MAPPING));
+        OMElement keyStoreMappingsElem = null;
+        IdentityConfigParser configParser = IdentityConfigParser.getInstance();
+        if (configParser != null) {
+            OMElement securityElem = configParser.getConfigElement(
+                    IdentityKeyStoreResolverConstants.CONFIG_ELEM_SECURITY);
+            if (securityElem != null) {
+                keyStoreMappingsElem = securityElem.getFirstChildWithName(
+                        IdentityKeyStoreResolverUtil.getQNameWithIdentityNameSpace(
+                                IdentityKeyStoreResolverConstants.CONFIG_ELEM_KEYSTORE_MAPPING));
+            }
+        }
+
+        if (keyStoreMappingsElem == null) {
+            LOG.warn(String.format("%s.%s element not found in identity.xml file. " +
+                            "You may ne using an outdated or  identity.xml.j2 file.",
+                    IdentityKeyStoreResolverConstants.CONFIG_ELEM_SECURITY,
+                    IdentityKeyStoreResolverConstants.CONFIG_ELEM_KEYSTORE_MAPPING));
+            return;
```

Comment:
```suggestion
            LOG.warn(String.format("The element %s.%s is not found in identity.xml file. ",                    IdentityKeyStoreResolverConstants.CONFIG_ELEM_SECURITY,
                    IdentityKeyStoreResolverConstants.CONFIG_ELEM_KEYSTORE_MAPPING));
            return;
```

Reply:
Fixed in https://github.com/wso2/carbon-identity-framework/pull/5978/commits/ef551e1bb2a5d092d74f0b968eb87df4a952ca20